sparse/impl/KokkosSparse_gauss_seidel_impl.hpp

```diff
@@ -1079,15 +1086,17 @@ class PointGaussSeidel {
           size_type num_large_rows = 0;
           KokkosSparse::Impl::kk_reduce_numrows_larger_than_threshold<
               row_lno_persistent_work_view_t, MyExecSpace>(
-              brows, permuted_xadj, num_values_in_l1, num_large_rows);
+              brows, permuted_xadj, num_values_in_l1, num_large_rows,
+              my_exec_space);
           num_big_rows = KOKKOSKERNELS_MACRO_MIN(
               num_large_rows,
-              (size_type)(MyExecSpace().concurrency() / suggested_vector_size));
+              (size_type)(my_exec_space.concurrency() / suggested_vector_size));
           // std::cout << "num_big_rows:" << num_big_rows << std::endl;
 
           if (KokkosKernels::Impl::kk_is_gpu_exec_space<MyExecSpace>()) {
             // check if we have enough memory for this. lower the concurrency if
             // we do not have enugh memory.
+            // TODO: Need to account for number of streams here?
```

Comment:
For the shared memory byte calculations, we may need to update the kk routines that query the exe space. If we can do that, I think the implementation can remain stream-agnostic.

Reply:
Actually the memory queries are independent of streams since the GPU memory is not split and allocated to each stream.

Reply:
Thanks for confirming